crypto/keys/pem_signer_factory.go

```diff
@@ -21,17 +21,32 @@ import (
 	"crypto/rsa"
 	"crypto/x509"
 	"fmt"
+	"sync"
 
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/any"
 	"github.com/google/trillian"
 	"github.com/google/trillian/crypto/keyspb"
 )
 
+var (
+	pkcs11Module string
+	pMu          sync.Mutex
+)
+
+// SetPKCS11Module should be called before creating a PEMSignerFactory
+// if the tree is expected to contain a PKCS#11 key
+func SetPKCS11Module(modulePath string) {
+	pMu.Lock()
+	defer pMu.Unlock()
+	pkcs11Module = modulePath
+}
+
 // PEMSignerFactory handles PEM-encoded private keys.
```

Comment:
I'd be ok with this being a follow-up PR. The interface is definitely still needed, because there are other implementations of SignerFactory outside of this repo.